python/hail/docs/guides/genetics.rst

```diff
@@ -0,0 +1,129 @@
+Genetics
+========
+
+Formatting
+~~~~~~~~~~
+
+Convert variants in string format to separate locus and allele fields
+.....................................................................
+
+..
+    >>> ht = hl.import_table('data/variant-lof.tsv')
+    >>> ht = ht.transmute(variant = ht.v)
+
+>>> ht = ht.key_by(**hl.parse_variant(ht.variant))
+
+If your variants are strings of the format 'chr:pos:ref:alt', you may want
+to convert them to separate locus and allele fields. This is useful if
+you have imported a table with variants in string format and you would like to
+join this table with other Hail tables that are keyed by locus and
+alleles.
+
+``hl.parse_variant(ht.variant)`` constructs a :class:`.StructExpression`
+containing two nested fields for the locus and alleles. The ** syntax unpacks
+this struct so that the resulting table has two new fields, ``locus`` and
+``alleles``.
+
+Filtering and Pruning
+~~~~~~~~~~~~~~~~~~~~~
+
+Remove related individuals from a dataset
+.........................................
+
+>>> pc_rel = hl.pc_relate(mt.GT, 0.001, k=2, statistics='kin')
+>>> pairs = pc_rel.filter(pc_rel['kin'] > 0.125)
+>>> related_samples_to_remove = hl.maximal_independent_set(pairs.i, pairs.j,
+...                                                        keep=False)
+>>> result = mt.filter_cols(
+...     hl.is_defined(related_samples_to_remove[mt.col_key]), keep=False)
+
+To remove related individuals from a dataset, we first compute a measure
+of relatedness between individuals using :func:`.pc_relate`. We filter this
+result based on a kinship threshold, which gives us a table of related pairs.
+
+From this table of pairs, we can compute the complement of the maximal
+independent set using :func:`.maximal_independent_set`. The parameter
+``keep=False`` in ``maximal_independent_set`` specifies that we want the
+complement of the set (the variants to remove), rather than the maximal
+independent set itself. It's important to use the complement for filtering,
+rather than the set itself, because the maximal independent set will not contain
+the singleton individuals.
+
+Once we have a list of samples to remove, we can filter the columns of the
+dataset to remove the related individuals.
+
+Filtering loci by a list of intervals
+.....................................
+
+>>> interval_table = hl.import_locus_intervals('data/gene.interval_list')
+>>> filtered_mt = mt.filter_rows(hl.is_defined(interval_table[mt.locus]))
+
+We have a matrix table ``mt`` containing the loci we would like to filter, and a
```

Comment:
I have the nagging sense that this is too much text / too much "understanding" for the how-to guide.

I don't want people to feel like they need to write a whole explanation of what's going on in order to contribute here.

I think this section should look something like:

```
Filtering from a Python list:
< example here, maybe using [hl.parse_interval(x) for x in list]

Importing intervals from a file:
<create intervals table>

Filtering if the table is very small:
<collect method>

Filtering if the table is large:
< hl.is_defined method>
```